src/Data/List/Lazy.purs

```diff
@@ -629,10 +629,10 @@ zip = zipWith Tuple
 --------------------------------------------------------------------------------
 
 instance showList :: (Show a) => Show (List a) where
-  show xs = "fromStrict (" ++ go (step xs) ++ ")"
-    where
-    go Nil = "Nil"
-    go (Cons x xs) = "Cons (" ++ show x ++ ") (" ++ go (step xs) ++ ")"
+  show xs = "toList " <> show (fromList truncated :: Array a) <> ellipsis
+    where truncated = take 5 six
```

Comment:
Sorry, I'm going to be a pain here. I just thought about this a bit more and realized, this way, there's no way to `show` more than 5 elements. Maybe we should create another function `showPrefix :: forall a. (Show a) => Int -> List a -> String`? That way, `show` is unsafe on infinite lists, but things stay flexible.